### ISSUE TITLE
Add a conditional helper for checking quarto profile value

### ISSUE THREAD

**Issue by cderv** [yihui/knitr#2216]
For quarto document, some conditional content can be made on profile value. When a profile is activated `QUARTO_PROFILE` env var is set. https://quarto.org/docs/projects/profiles.html#overview

It would be interesting to have a helper function like for output format (`is_html_output()`) to test profile value 

This would allow conditional chunk based on quarto profile value. 

It is easy to do `eval = Sys.getenv("QUARTO_PROFILE") == "profile name"` but we could offer a nice shorter version. 

We could either have a `is_profile("profile name")` that would be true on specific profile. Or another name more namespaced (`quarto_profile()` ? `is_quarto_profile()` ? `when_quarto_profile()` ? )

**Comment by yihui** [comment]
Thanks @cderv. Added `quarto_profile()`: with no argument it returns the active profile name(s) read from `QUARTO_PROFILE` (comma-split and trimmed, so multi-profile setups work), and given one or more names it returns whether any is active. So a profile-conditional chunk is just `eval = knitr::quarto_profile('production')`.

**Comment by cderv** [comment]
Oh thank you ! I am thinking as time passed since I opened this issue that we could also have made this function into quarto R 📦 too if you prefer. It is just convenient to have it in the same place as `is_pandoc()` or `is_html_output()`. It would have been `eval = quarto::quarto_profile('production').`

Happy to keep it in knitr though. Thanks.

**Comment by yihui** [comment]
Yes, I also thought about that, and I have no opinion on whether this function is hosted in **knitr** or **quarto**. I put it in **knitr** just because the `pandoc_to()` function is similar to `quarto_profile()`.
